pkg/dbutil/common.go

```diff
@@ -617,3 +633,110 @@ func ReplacePlaceholder(str string, args []string) string {
 	newStr := strings.Replace(str, "?", "'%s'", -1)
 	return fmt.Sprintf(newStr, utils.StringsToInterfaces(args)...)
 }
+
+// ExecSQLWithRetry executes sql with retry
+func ExecSQLWithRetry(ctx context.Context, db *sql.DB, sql string, args ...interface{}) (err error) {
+	for i := 0; i < DefaultRetryTime; i++ {
+		startTime := time.Now()
+		_, err = db.ExecContext(ctx, sql, args...)
+		if err == nil {
+			takeDuration := time.Since(startTime)
+			if takeDuration > SlowWarnLog {
+				log.Warn("exec sql slow", zap.String("sql", sql), zap.Reflect("args", args), zap.Duration("take", takeDuration))
+			}
+			return nil
+		}
+
+		if ignoreError(err) {
```

Comment:
move this block before L642, add reset err to `nil` (`err = nil`) in it, so we can reuse the `SlowWarnLog` mechanism？

Reply:
updated in 06ee87d, print slow log before judge error. do you think it's ok ?

Reply:
fine